compiler/damlc/tests/daml-test-files/SemanticsEvalOrder.daml

```diff
@@ -11,116 +11,139 @@
 -- `evExpRecUpdErr1`, `evExpRecUpdErr2_1`, and `evExpRecUpdErr2_2`, for example.)
 
 {-# OPTIONS_GHC -Wno-controller-can #-}
+{-# LANGUAGE ApplicativeDo #-}
 
 module SemanticsEvalOrder where
```

Comment:
Not really sure we need those test. 
We already test that extensively in [Scala ](https://github.com/digital-asset/daml/blob/main/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/EvaluationOrderTest.scala).


What your opinion @basvangijzel-DA @akrmn  ?

Reply:
until we know they're unused, I'd rather keep the tests here that correspond to general eval semantics, while removing/commenting the ones specifically for `Scenario` eval semantics